src/fsharp/FSharp.Core/seq.fsi

```diff
@@ -683,11 +850,41 @@ module Seq =
     /// <param name="state">The initial state.</param>
     ///
     /// <returns>The state object after the folding function is applied to each element of the sequence.</returns>
+    ///
     /// <exception cref="T:System.ArgumentNullException">Thrown when the input sequence is null.</exception>
     ///
     /// <remarks>This function consumes the whole input sequence before returning the result.</remarks>
-    /// 
-    /// <example-tbd></example-tbd>
+    ///
+    /// <example id="fold-back-1">
+    /// <code lang="fsharp">
+    /// type Count =
+    ///   { Positive: int
+    ///     Negative: int
+    ///     History: int list }
+    ///
+    /// ( [1; 0; -1; -2; 3],
+    ///   {Positive = 0; Negative = 0; History = []})
+    /// ||> Seq.foldBack
+    ///       (fun a acc  ->
+    ///         let history = acc.History @ [a]
```

Comment:
This example is a bit problematic because it is quadratic  - `... @ [a]` is a bit of a code-smell because of this.  

I mean, it's ok since the input list is so small.  Just mentioning this in case you think of another way of doing this that doesn't use append-to-end

Reply:
I could use a mutable list or change the example altogether and use a stringbuilder to create a history. I would like to know how you would build an immutable list in a similar situation.